src/SourceGenerators/Uno.UI.SourceGenerators/Helpers/SymbolExtensions.cs

```diff
@@ -6,18 +6,100 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
-using System.Threading.Tasks;
 
 namespace Microsoft.CodeAnalysis
 {
-    /// <summary>
-    /// Roslyn symbol extensions
-    /// </summary>
-    internal static class SymbolExtensions
+	/// <summary>
+	/// Roslyn symbol extensions
+	/// </summary>
+	internal static class SymbolExtensions
 	{
 		private static bool IsRoslyn34OrEalier { get; }
 			= typeof(INamedTypeSymbol).Assembly.GetVersionNumber() <= new Version("3.4");
 
+
+		/// <summary>
+		/// Given an <see cref="INamedTypeSymbol"/>, add the symbol declaration (including parent classes/namespaces) to the given <see cref="IIndentedStringBuilder"/>.
+		/// </summary>
+		/// <remarks>
+		/// <para>IMPORTANT: The returned stack must be disposed after putting everything for the given <see cref="INamedTypeSymbol"/>.</para>
+		/// <para>Example usage:</para>
+		/// <code><![CDATA[
+		/// var stack = myClass.AddToIndentedStringBuilder(builder);
+		/// using (builder.BlockInvariant("public static void M()"))
+		/// {
+		///     builder.AppendLineInvariant("Console.WriteLine(\"Hello world\")");
+		/// }
+		/// 
+		/// while (disposables.Count > 0)
+		/// {
+		///     disposables.Pop().Dispose();
+		/// }
+		/// ]]></code>
+		/// <para>NOTE: Another possible implementation is to accept an <see cref="Action"/> as a parameter to generate the type members, execute the action here, and also dispose
+		/// the stack here. The advantage is that callers don't need to worry about disposing the stack.</para>
+		/// </remarks>
+		public static Stack<IDisposable> AddToIndentedStringBuilder(this INamedTypeSymbol namedTypeSymbol, IIndentedStringBuilder builder, Action<IIndentedStringBuilder>? beforeClassHeaderAction = null)
```

Comment:
Hopefully, this should make it easy for other generators to prepare the boilerplate code.